src/lib/message/L2ToL1Message.ts

```diff
@@ -72,32 +72,28 @@ export type L2ToL1MessageReaderOrWriter<T extends SignerOrProvider> =
 export class L2ToL1Message {
   public static fromEvent<T extends SignerOrProvider>(
     l1SignerOrProvider: T,
-    event?: L2ToL1TransactionEvent['args'],
-    outboxAddress?: string,
-    batchNumber?: BigNumber,
-    indexInBatch?: BigNumber
+    event: L2ToL1TransactionEvent['args'],
+    outboxAddress: string,
```

Comment:
can't we infer the outbox address instead of having it user provided?
if its nitro, its the only outbox
if its pre-nitro, check the batchNum to get the outbox

Reply:
We dont have a network object here to lookup the outbox address on